src/benchmarklib/listenable.cpp

```diff
@@ -0,0 +1,22 @@
+#include "listenable.hpp"
+
+namespace opossum {
+
+void Listenable::add_listener(const Event event, EventListener listener) {
+  if (_event_listeners.find(event) == _event_listeners.end()) {
```

Comment:
If we were to use CRTP here, we could save us that map

Reply:
Hmm, not sure if that's right. We still need to map the specific Events to their listeners, right?